### ISSUE TITLE
It appears '$foo' and "\$foo" can both be prohibited under common(ish) configurations

### ISSUE THREAD

**Issue by shadowcat-mst** [Perl-Critic/Perl-Critic#977]
To be clear:

    my $string = '$foo';

runs afoul of

https://metacpan.org/pod/Perl::Critic::Policy::ValuesAndExpressions::RequireInterpolationOfMetachars

but

    my $string = "\$foo";

runs afoul of

https://metacpan.org/pod/Perl::Critic::Policy::ValuesAndExpressions::ProhibitInterpolationOfLiterals

which is ... suboptimal.

I'm not entirely sure what the right answer here is (other than "you're doing something unusual, explicitly nocritic one of the policies") though maybe the latter policy could have a flag for basically 'allow anything the former policy would refuse'? Whether that would be a net win or not is debatable, but it seems a bit strange for two policies that are both categorised qw(core pbp cosmetic) and therefore very easy to enable together to between them make it impossible to express something at all.

**Comment by shadowcat-mst** [comment]
Tagging in @lechner since this is also something we discovered through the lintian work and they're more likely to keep track of this bug than I am

**Comment by lechner** [comment]
Yeah, thanks! I had subscribed already. The first policy, [RequireInterpolationOfMetachars](https://metacpan.org/pod/Perl::Critic::Policy::ValuesAndExpressions::RequireInterpolationOfMetachars), actually found more issues than any other recently, so I am keen on adopting it.

Here is [one issue](https://salsa.debian.org/lintian/lintian/-/blob/master/private/refresh-mail-transport-agents#L26-27) I can't seem to get around.
